crates/jstz_cli/src/deploy.rs

```diff
@@ -1,25 +1,59 @@
+use dialoguer::Confirm;
 use jstz_proto::{
     operation::{Content, DeployFunction, Operation, SignedOperation},
     receipt::Content as ReceiptContent,
 };
 use log::{debug, info};
 
 use crate::{
+    account,
     config::{Config, NetworkName, SmartFunction},
     error::{bail, bail_user_error, user_error, Result},
+    sandbox::daemon,
+    term::styles,
     utils::read_file_or_input_or_piped,
 };
 
 pub async fn exec(
-    code: Option<String>,
+    code_op: Option<String>,
     balance: u64,
     name: Option<String>,
     network: Option<NetworkName>,
 ) -> Result<()> {
     let mut cfg = Config::load()?;
 
-    let (_, user) = cfg.accounts.current_user().ok_or(user_error!(
-        "You are not logged in. Please run `jstz login`."
+    // Load sandbox if network is Dev or it is None and cfg network is dev, and sandbox is not already loaded
+    if (network == Some(NetworkName::Dev)
+        || (network.is_none() && cfg.networks.default_network == Some(NetworkName::Dev)))
+        && cfg.sandbox.is_none()
+    {
+        info!("No sandbox is currently running.");
+
+        let proceed = Confirm::new()
+            .with_prompt(format!("Start the sandbox in daemon mode now? Tip: Use '{}' for an interactive session instead.", styles::command("jstz sandbox start")))
+            .default(true)
+            .interact()?;
+
+        if proceed {
+            // User confirmed, start the sandbox
+            daemon::main(true, false, &mut cfg).await?;
+            info!(
+                "Use `{}` to start from a clear sandbox state.\n",
+                styles::command("jstz sandbox restart --detach")
+            );
+        } else {
+            bail_user_error!(
+                "Please start the sandbox before deploying a smart function to dev network."
+            );
+        }
+    }
```

Comment:
I thought we agreed not to automatically start the sandbox?

Reply:
this doesn't start it automatically - it asks it with a message whether they want to start it in background and says how to start the interactive session with just "jstz sandbox start".

Reply:
I've removed the interaction so that it moves quicker, but I don't see any issue with it and would prefer adding it later at least during development as it makes testing much more pleasant.